tools/build_defs/framework.bzl

```diff
@@ -401,35 +399,33 @@ def _define_outputs(ctx, attrs, lib_name):
 
     _check_file_name(lib_name, "Library name")
 
-    installdir = ctx.actions.declare_directory(lib_name)
-    out_include_dir = ctx.actions.declare_directory(lib_name + "/" + attrs.out_include_dir)
-    out_bin_dir = ctx.actions.declare_directory(lib_name + "/" + attrs.out_bin_dir)
-    out_lib_dir = ctx.actions.declare_directory(lib_name + "/" + attrs.out_lib_dir)
+    out_binary_files = []
+    for file in attrs.binaries:
+        out_binary_files += [_declare_out(ctx, lib_name, attrs.out_bin_dir, file)]
 
-    out_binary_files = _declare_out(ctx, lib_name, out_bin_dir, attrs.binaries)
+    out_include_dir = ctx.actions.declare_directory(lib_name + "/" + attrs.out_include_dir)
+    installdir = '/'.join(out_include_dir.path.split('/')[:-1])
 
     libraries = LibrariesToLinkInfo(
-        static_libraries = _declare_out(ctx, lib_name, out_lib_dir, static_libraries),
-        shared_libraries = _declare_out(ctx, lib_name, out_lib_dir, attrs.shared_libraries),
-        interface_libraries = _declare_out(ctx, lib_name, out_lib_dir, attrs.interface_libraries),
+        static_libraries = _declare_out(ctx, lib_name, attrs.out_lib_dir, static_libraries),
+        shared_libraries = _declare_out(ctx, lib_name, attrs.out_lib_dir, attrs.shared_libraries),
+        interface_libraries = _declare_out(ctx, lib_name, attrs.out_lib_dir, attrs.interface_libraries),
     )
-    declared_outputs = [installdir, out_include_dir, out_bin_dir, out_lib_dir] + out_binary_files
+    declared_outputs = [out_include_dir] + out_binary_files
```

Comment:
Since we are keeping includedir as it was, we should include it back into the list